generators/client/templates/angular/src/main/webapp/app/shared/auth/_user-route-access-service.ts

```diff
@@ -1,15 +1,37 @@
 import { Injectable } from '@angular/core';
-import { CanActivate, Router, ActivatedRouteSnapshot } from '@angular/router';
+import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot } from '@angular/router';
 
 import { AuthService } from '../';
+import { Principal } from '../';
+import { LoginModalService } from '../login/login-modal.service';
+import { StateStorageService } from './state-storage.service';
 
 @Injectable()
 export class UserRouteAccessService implements CanActivate {
 
-    constructor(private router: Router, private auth: AuthService) {
+    constructor(private router: Router,
+                private loginModalService: LoginModalService,
+                private principal: Principal,
+                private stateStorageService: StateStorageService) {
     }
 
-    canActivate(route: ActivatedRouteSnapshot): boolean | Promise<boolean> {
-        return this.auth.authorize(false).then( canActivate => canActivate);
+    canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean | Promise<boolean> {
+        return this.checkLogin(route.data['authorities'], state.url);
     }
+
+    checkLogin(authorities: string[], url: string): boolean {
```

Comment:
guess this will always return false, no matter what. 

can you return promise<Boolean>

something like this

```js
return promise.resolve(this.principal.hasAnyAuthority......
```